packages/visx-xychart/src/components/axis/BaseAxis.tsx

```diff
@@ -35,15 +36,19 @@ export default function BaseAxis<Scale extends AxisScale>({
 
   const tickLabelProps = useMemo(
     () =>
-      props.tickLabelProps || // use props if passed
-      (axisStyles // else construct from theme if possible
-        ? () =>
-            orientation === 'left' || orientation === 'right'
-              ? // by default, wrap vertical-axis tick labels within the allotted margin space
-                // this does not currently account for axis label
-                { ...axisStyles.tickLabel, width: margin?.[orientation] }
-              : axisStyles.tickLabel
-        : undefined),
+      props.tickLabelProps || axisStyles // construct from props + theme if possible
+        ? (value: ScaleInput<Scale>, index: number) =>
+            // by default, wrap vertical-axis tick labels within the allotted margin space
+            // this does not currently account for axis label
+            ({
+              ...axisStyles?.tickLabel,
+              width:
+                orientation === 'left' || orientation === 'right'
+                  ? margin?.[orientation]
+                  : undefined,
+              ...props.tickLabelProps?.(value, index),
```

Comment:
the only thing that was changed here is that all `...axisStyles.tickLabel` styles are applied by default and the user-specified styles (`props.tickLabelProps`) are overrides. this makes it way easier to override a single style or two.